src/HTTPClient.ts

```diff
@@ -27,10 +27,41 @@ export class HTTPClient {
 	 * @public
 	 * @constructor
 	 * Creates an instance of a HTTPClient
-	 * @param {Middleware} middleware - The first middleware of the middleware chain
+	 * @param {Middleware| Middleware[]} middleware - The first middleware of the middleware chain or the array of middleware handlers
 	 */
-	public constructor(middleware: Middleware) {
-		this.middleware = middleware;
+	public constructor(middleware: Middleware | Middleware[]) {
+		this.setMiddleware(middleware);
+	}
+
+	/**
+	 * @private
+	 * Verifies if the middleware passed is an array or not before setting this.middleware property
+	 * @param {any} middleware - The middleware passed
+	 * @returns Nothing
+	 */
+
+	private setMiddleware(middleware: any): void {
+		if (Object.prototype.toString.call(middleware) === "[object Array]") {
```

Comment:
```suggestion
		if (middleware.length > 1) {
```

Reply:
> I feel the condition `Object.prototype.toString.call(middleware) === "[object Array]"`  indicates more clearly the purpose.

Reply:
that's a debatable opinion.
I think we should focus more on trying to keep as much as we can a strongly type API surface at every level to get compile time validation and more. `Middleware: any` is problematic here as it "looses" some of the compile time validation (both for the user of the method and for the implementation).
On top of that varargs are common practice in javascript/typescript allowing you to provide a clean, strongly typed API surface which serves multiple scenarios (single or multiple arguments)

Reply:
Using a spread operator requires more changes which I would like to take care of in a follow up PR.